### PR TITLE
Add very basic test for global add

### PR DESCRIPTION
**Pull request by torifat**
**Summary**

Add very basic test for `global add` which covers #1840

_I will try to cover the rest as soon as I can._

### CODE REVIEW

**Comment by torifat** [comment]
@wyze Can you review this please? I need this to write test for #1889

File: __tests__/commands/global.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    await config.init({
+      cwd,
+      globalFolder: path.join(cwd, '.yarn/.global'),
+      cacheFolder: path.join(cwd, '.yarn'),
+      linkFolder: path.join(cwd, '.yarn/.link'),
+    });
+
+    await global(config, reporter, flags, [command, ...args]);
+
+    if (checkGloabl) {
```
Comment (wyze) [inline __tests__/commands/global.js]: Fix misspelling of Global here and the other places.

File: __tests__/commands/global.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      await checkGloabl(config, reporter, out);
+    }
+
+  } catch (err) {
+    throw new Error(`${err && err.stack} \nConsole output:\n ${out}`);
+  }
+}
+test.concurrent('add without flag', (): Promise<void> => {
+  return runGlobal('add', {}, ['react-native-cli'], 'add-without-flag');
```
Comment (wyze) [inline __tests__/commands/global.js]: Should there be an assertion here to verify the module was added?

File: __tests__/commands/global.js (hunk 1 above)
Comment (torifat) [inline __tests__/commands/global.js]: Ah, Thanks 👍

File: __tests__/commands/global.js (hunk 2 above)
Comment (torifat) [inline __tests__/commands/global.js]: Sadly it has another issue so I skipped the assertion for the time being.

https://github.com/yarnpkg/yarn/blob/master/src/cli/commands/global.js#L35

This resets the config obejct. So, It is difficult to test things after the command has executed. The returned `config` object no more contains the value we set for `globalFolder`.

Comment (wyze) [inline __tests__/commands/global.js]: Alright.

Comment (torifat) [inline __tests__/commands/global.js]: I can assert `out` though. Going for it now.

Comment (torifat) [inline __tests__/commands/global.js]: Never mind, I thought `out` is `Object` but it's string 😢

Comment (wyze) [inline __tests__/commands/global.js]: If you switch it to BufferReporter, you could do `reporter.getBuffer()` that will give you an array of output.

Comment (wyze) [inline __tests__/commands/global.js]: See the tests here as an example: https://github.com/yarnpkg/yarn/blob/master/__tests__/commands/ls.js

Comment (torifat) [inline __tests__/commands/global.js]: Using `config.cwd`. Things for the idea though. I might use it for something else later 😄

**Comment by torifat** [comment]
@wyze Travis is not working 😞

**Comment by wyze** [comment]
Travis is having issues with the NodeSource Debian repository.

**Comment by onemen** [comment]
The test add bins links to the user bin folder `yarn global bin` instead of the test folder

**Comment by torifat** [comment]
@onemen can you please send a PR with the fix?

**Comment by onemen** [comment]
Done #1918

**Pull request merged**